src/Context/TodoContext.tsx

```diff
@@ -0,0 +1,205 @@
+  error: CurrentError,
+  setError: (error: CurrentError) => void;
+  handleToggleChange: (todo: Todo) => void;
+  handleTodoDelete: (id: number) => void;
+  handleTodoAdd: (newTodo: Omit<Todo, 'id'>) => Promise<void>
+  handleTodoRename: (todo: Todo, newTodoTitle: string) => Promise<void>,
+  completedTodos: Todo[];
+  activeTodos: Todo[];
+  handleClearCompleted: () => void;
+  processingTodoIds: number[];
+  setProcessingTodoIds: (todoIdsToDelete: number[]) => void;
+}
+
+const initalContext: TodoContextInterface = {
+  todos: [],
+  setTodos: () => {},
+  tempTodo: null,
+  setTempTodo: () => {},
+  isLoading: false,
+  setIsLoading: () => {},
+  error: CurrentError.Default,
+  setError: () => {},
+  handleToggleChange: () => {},
+  handleTodoDelete: () => {},
+  handleTodoAdd: async () => {},
+  handleTodoRename: async () => {},
+  completedTodos: [],
+  activeTodos: [],
+  handleClearCompleted: () => {},
+  processingTodoIds: [],
+  setProcessingTodoIds: () => {},
+};
+
+export const TodoContext = createContext(initalContext);
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [error, setError] = useState(CurrentError.Default);
+  const [isLoading, setIsLoading] = useState(false);
+  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([]);
+
+  const completedTodos = getCompletedTodos(todos);
+  const activeTodos = getActiveTodos(todos);
+
+  const handleTodoDelete = (todoId: number) => {
+    setIsLoading(true);
+    setProcessingTodoIds(prevState => [...prevState, todoId]);
+
+    todoService.deleteTodo(todoId)
+      .then(() => {
+        setTodos(prevTodos => {
+          return prevTodos.filter(todo => todo.id !== todoId);
+        });
+      })
+      .catch(() => {
+        setError(CurrentError.DeleteError);
+        throw new Error();
+      })
+      .finally(() => {
+        setProcessingTodoIds(
+          (prevState) => prevState.filter(id => id !== todoId),
+        );
+        setIsLoading(false);
+      });
+  };
+
+  const handleTodoAdd = (newTodo: Omit<Todo, 'id'>) => {
+    setIsLoading(true);
+
+    return todoService.addTodo(newTodo)
+      .then(createdTodo => {
+        setTodos((prevTodos) => [...prevTodos, createdTodo]);
+      })
+      .catch(() => {
+        setError(CurrentError.AddError);
+        throw new Error();
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const handleTodoRename = (todo: Todo, newTodoTitle: string) => {
+    if (todo.title === newTodoTitle) {
+      return Promise.resolve();
+    }
+
+    if (!newTodoTitle.trim()) {
+      setError(CurrentError.EmptyTitleError);
+
+      return Promise.reject();
+    }
+
+    setIsLoading(true);
+    setProcessingTodoIds(prevState => [...prevState, todo.id]);
+
+    // eslint-disable-next-line consistent-return
```

Comment:
Don't disable eslint, I think in this case you  actually don't need to return

Reply:
If I don't have return, then when you edit todo, after edit on load it shows the previous version of the title and then shows the new one (after load)

but with return, everything works as it should...